tasks/eva_2406/gather_release_species.py

```diff
@@ -0,0 +1,357 @@
+    or that are human and therefore cannot be released
+    """
+    for assembly, tax_id, study in query_results:
+        if not assembly or not tax_id:
+            logger.error('Study %s is missing assembly (%s) or taxonomy id (%s)', study, assembly, tax_id)
+        elif tax_id == 9606:
+            logger.debug("Study %s is human and won't be releases", study)
+        else:
+            yield assembly, tax_id, study
+
+
+def parse_dbsnp_csv(input_file, accession_counts):
+    """Parse the CSV file generated in the past year to get the DBSNP data"""
+    df = pd.read_csv(input_file)
+    df = df[df.Source != 'EVA']
+    taxids = []
+    scientific_names = []
+    ensembl_assemblies_from_taxid = []
+    ensembl_assemblies_from_assembly = []
+    target_assemblies = []
+
+    for index, record in df.iterrows():
+        taxid, scientific_name, ensembl_assembly_from_taxid = retrieve_current_ensembl_assemblies(record['Taxid'])
+        _, _, ensembl_assembly_from_assembly = retrieve_current_ensembl_assemblies(record['Assembly'])
+        taxids.append(taxid)
+        scientific_names.append(scientific_name)
+        ensembl_assemblies_from_taxid.append(ensembl_assembly_from_taxid)
+        ensembl_assemblies_from_assembly.append(ensembl_assembly_from_assembly)
+        target_assemblies.append(ensembl_assembly_from_taxid or ensembl_assembly_from_assembly)
+        if record['Assembly'] != 'Unmapped':
+            _, _, count = accession_counts[record['Assembly']]
+            if count != int(record['Number Of Variants (submitted variants)'].replace(',', '')):
+                logger.error(
+                    'Count in spreadsheet (%s) and in database (%s) are different for accession %s',
+                    record['Number Of Variants (submitted variants)'], count, record['Assembly']
+                )
+    df['Scientific Name'] = scientific_names
+    df['Ensembl assembly from taxid'] = ensembl_assemblies_from_taxid
+    df['Ensembl assembly from assembly'] = ensembl_assemblies_from_assembly
+    df['Target Assembly'] = target_assemblies
+    df.replace(',', '', regex=True, inplace=True)
+    return df
+
+
+def create_table_for_progress(private_config_xml_file):
+    with psycopg2.connect(get_pg_metadata_uri_for_eva_profile("development", private_config_xml_file),
+                          user="evadev") as metadata_connection_handle:
+        query_create_table = (
+            'CREATE TABLE IF NOT EXISTS remapping_progress '
+            '(source TEXT, taxid INTEGER, scientific_name TEXT, assembly_accession TEXT, number_of_study INTEGER NOT NULL,'
+            'number_submitted_variants BIGINT NOT NULL, release_number INTEGER, `target_assembly_accession` TEXT, '
+            'report_time TIMESTAMP DEFAULT NOW(), progress_status TEXT, start_time TIMESTAMP, '
+            'completion_time TIMESTAMP, '
+            'primary key(source, taxid, assembly_accession, release_number))'
```

Comment:
@sundarvenkata-EBI Could you let me know if you think this is enough information to record the remapping process?

Reply:
I would suggest adding the remapping pipeline version too...

Reply:
I thought we would put the metric in another tables alongside the versions but we might as well put them there. 
I've added columns for both.